packages/foundation-models/README.md

```diff
@@ -17,9 +17,153 @@ $ npm install @sap-ai-sdk/foundation-models
   - Create a `.env` file in the sample-code directory.
   - Add an entry `AICORE_SERVICE_KEY='<content-of-service-key>'`.
 
-## Usage
+## Azure OpenAI Client
 
-<!-- Details to be added -->
+To make a generative AI model available for use, you need to create a deployment.
+You can create a deployment for each model and model version, as well as for each resource group that you want to use with generative AI hub.
+
+After the deployment is complete, you have a `deploymentUrl`, which can be used to access the model.
+
+The Azure OpenAI client allows you to send chat completion or embedding requests to OpenAI models deployed in SAP generative AI hub.
+
+### Prerequisites
+
+- A deployed OpenAI model in SAP generative AI hub.
+  - You can use the [`DeploymentApi`](../ai-api/README.md#deploymentapi) from `@sap-ai-sdk/ai-api` to deploy a model to SAP generative AI hub. For more information, see [here](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/create-deployment-for-generative-ai-model-in-sap-ai-core)
+- `sap-ai-sdk/foundation-models` package installed in your project.
+
+### Usage of Azure OpenAI Chat Client
+
+Use the `AzureOpenAiChatClient` to send chat completion requests to an OpenAI model deployed in SAP generative AI hub.
+You can pass the model name as a parameter to the client, the sdk will implicitly fetch the deployment ID for the model from the AI Core service and use it to send the request.
+
+The deployment information which includes deployment ID and properties like model name and model version is also cached by default for 5 mins so that performance is not impacted by fetching the deployment information for every request.
+
+```TS
+import { AzureOpenAiChatClient } from '@sap-ai-sdk/foundation-models';
+
+const client = new AzureOpenAiChatClient('gpt-35-turbo');
+const response = await client.run({
+  messages: [
+    {
+      role: 'user',
+      content: 'Where is the deepest place on earth located'
+    }
+  ]
+});
+
+const responseContent = response.getContent();
+
+```
+
+To send a chat completion request with system messages use:
+
+```TS
+import { AzureOpenAiChatClient } from '@sap-ai-sdk/foundation-models';
+
+const client = new AzureOpenAiChatClient('gpt-35-turbo');
+const response = await client.run({
+  messages: [
+    {
+      role: 'system',
+      content: 'You are a friendly chatbot.'
+    },
+    {
+      role: 'user',
+      content: 'Hi, my name is Isa'
+    },
+    {
+      role: 'system',
```

Comment:
```suggestion
      role: 'assistant',
```

I think one typically only has one system message, this message rather looks like a previous response from an LLM

Reply:
Hmm, I didn't know this. But you are right all examples have only one system message.